### PR TITLE
Fix auto-registration of child device op dirs #2389

### PR DESCRIPTION
**Pull request by albinsuresh**
## Proposed changes

If the child device directory the child device operation directories follow the default external id naming scheme, they are auto-registered with the right topic id and name derived from that external id.

## Types of changes

- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Improvement (general improvements like code refactoring that doesn't explicitly fix a bug or add any new functionality)
- [ ] Documentation Update (if none of the other choices apply)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Paste Link to the issue

https://github.com/thin-edge/thin-edge.io/issues/2389

## Checklist

- [x] I have read the [CONTRIBUTING](https://github.com/thin-edge/thin-edge.io/blob/main/CONTRIBUTING.md) doc
- [x] I have signed the [CLA](https://github.com/thin-edge/thin-edge.io/blob/main/CONTRIBUTOR-LICENSE-AGREEMENT.md) (in all commits with git commit -s)
- [x] I ran `cargo fmt` as mentioned in [CODING_GUIDELINES](https://github.com/thin-edge/thin-edge.io/blob/main/CODING_GUIDELINES.md)
- [x] I used `cargo clippy` as mentioned in [CODING_GUIDELINES](https://github.com/thin-edge/thin-edge.io/blob/main/CODING_GUIDELINES.md)
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

### CODE REVIEW

File: crates/extensions/c8y_mapper_ext/src/converter.rs [hunk 1]
```diff
@@ -752,14 +752,15 @@ impl CumulocityConverter {
                     continue;
                 }
             };
-            let child_topic_id =
-                EntityTopicId::default_child_device(child_external_id.as_ref()).unwrap();
+
+            let child_name = self.default_device_name_from_external_id(&child_external_id);
+            let child_topic_id = EntityTopicId::default_child_device(&child_name).unwrap();
```
Comment (didier-wenzek) [inline crates/extensions/c8y_mapper_ext/src/converter.rs:757]: This `unwrap()` has to be removed as a maliciously crafted device name will crash the mapper.

**Review by didier-wenzek — APPROVED** [review]
Approved

File: crates/extensions/c8y_mapper_ext/src/converter.rs (hunk 1 above)
Comment (didier-wenzek) [reply crates/extensions/c8y_mapper_ext/src/converter.rs:757]: Not urgent. The error is catched before.

```
Supported operations of child device directory: demo-device-777:device:+ ignored due to Invalid external ID: demo-device-777:device:+ contains invalid character: +    
```

Comment (albinsuresh) [reply crates/extensions/c8y_mapper_ext/src/converter.rs]: Fixed

**Review by reubenmiller — APPROVED** [review]
Approved. I also added some system tests

**Pull request merged**